### PR TITLE
Add test for non-intuitive merkle tree gadget witnessing behavior.

### LINKED ISSUE #920: Test that `merkle_tree_check_read_gadget` behavior with respect to `read_success` acts as expected
Write a test that ensures the `merkle_tree_check_read_gadget` behaves properly, in particular, during witnessing there is a weird API edge case that must be examined. (See [this comment](https://github.com/zcash/zcash/issues/920#issuecomment-233811236).)

### PR DESCRIPTION
**Pull request by ebfull**
Closes #920

### CODE REVIEW

File: src/gtest/test_circuit.cpp [hunk 1]
```diff
@@ -74,3 +77,108 @@ TEST(circuit, endianness)
...
+
+    // Test the case where the first commitment is enforced but the second isn't
+    // Works because the first read is performed before the second one
+    ASSERT_TRUE(test_merkle_gadget<FieldT>(true, false, false));
+    ASSERT_TRUE(test_merkle_gadget<FieldT>(true, false, true));
+
+    // Test the case where the first commitment isn't enforced but the second is
+    // Doesn't work because the first multipacker witnesses the existing root (which
+    // is null)
+    ASSERT_TRUE(!test_merkle_gadget<FieldT>(false, true, false));
```
Comment (defuse) [inline src/gtest/test_circuit.cpp]: Is there an ASSERT_FALSE? (I was about to write "Should this be ASSERT_FALSE?" then I noticed the `!`).

**Comment by defuse** [comment]
ACK!

**Comment by ebfull** [comment]
@zkbot r+

**Comment by daira** [comment]
Post-hoc ACK.

**Comment by daira** [comment]
See also #1136.
